File: history_today-master/app/src/main/java/com/animee/todayhistory/ui/HistoryDescActivity.java
package com.animee.todayhistory.ui;

import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.util.Log;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.animee.todayhistory.R;
import com.animee.todayhistory.bean.HistoryDescBean;
import com.squareup.picasso.Picasso;

public class HistoryDescActivity extends androidx.appcompat.app.AppCompatActivity implements View.OnClickListener, HistoryDescContact.View{
    private String TAG = "HistoryDescActivity";
    private ImageView backIv,shareIv,picIv;
    private TextView titleTv;
    private TextView contentTv;
    private String id;

    HistoryDescBean.ResultBean resultBean;
    HistoryDescPresenter presenter;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_history_desc);
        backIv    = findViewById(R.id.desc_back_iv);
        shareIv   = findViewById(R.id.desc_share_iv);
        picIv     = findViewById(R.id.desc_iv_pic);
        titleTv   = findViewById(R.id.desc_tv_title);
        contentTv = findViewById(R.id.desc_tv_content);
        backIv.setOnClickListener(this);
        shareIv.setOnClickListener(this);

        id = getIntent().getStringExtra("hisId");
        presenter = new HistoryDescPresenter(this);
        presenter.loadGet(id);
    }
    
    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (presenter != null) {
            presenter.onDestroy();
        }
    }


    @Override
    public void onClick(View v) {
        if (v.getId() == R.id.desc_back_iv) {
            finish();
        } else if (v.getId() == R.id.desc_back_iv) {
            String text = "我发现一款好用的软件-历史上的今天，快来一起探索这个APP吧！";
            if (resultBean != null) {
                text = "想要了解" + resultBean.getTitle() + "详情么？快来下载历史上的今天App吧！";
            }
            Intent intent = new Intent(Intent.ACTION_SEND);
            intent.setType("text/plain");
            intent.putExtra(Intent.EXTRA_TEXT, text);
            startActivity(Intent.createChooser(intent, "历史上的今天"));
        }
    }



    @Override
    public void showResult(HistoryDescBean bean) {
        if (bean.getResult() != null && !bean.getResult().isEmpty()) {
            resultBean = bean.getResult().get(0);

            final HistoryDescActivity historyDescActivity = this;

            Log.d(TAG, String.valueOf(resultBean));
            titleTv.setText(resultBean.getTitle());
            contentTv.setText(resultBean.getContent());
            
            // Handle pictures - new structure has picUrl list
            if (resultBean.getPicUrl() != null && !resultBean.getPicUrl().isEmpty()) {
                String firstPicUrl = resultBean.getPicUrl().get(0).getUrl();
                if (!TextUtils.isEmpty(firstPicUrl)) {
                    picIv.setVisibility(View.VISIBLE);
                    Picasso.get().load(firstPicUrl).into(picIv);
                } else {
                    picIv.setVisibility(View.GONE);
                }
            } else {
                picIv.setVisibility(View.GONE);
            }
        } else {
            // Handle empty result case
            titleTv.setText("暂无数据");
            contentTv.setText("未找到相关历史事件详情");
            picIv.setVisibility(View.GONE);
        }

    }




}

File: history_today-master/app/src/main/java/com/animee/todayhistory/ui/HistoryDescActivity.kt
package com.animee.todayhistory.ui

import android.content.Intent
import android.os.Bundle
import android.text.TextUtils
import android.util.Log
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.animee.todayhistory.R
import com.animee.todayhistory.bean.HistoryDescBean
import com.squareup.picasso.Picasso

class HistoryDescActivity : AppCompatActivity(), View.OnClickListener, HistoryDescContact.View {

    private val TAG = "HistoryDescActivity"

    private lateinit var backIv: ImageView
    private lateinit var shareIv: ImageView
    private lateinit var picIv: ImageView
    private lateinit var titleTv: TextView
    private lateinit var contentTv: TextView
    private var resultBean: HistoryDescBean.ResultBean? = null
    private var presenter: HistoryDescPresenter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_history_desc)

        backIv = findViewById(R.id.desc_back_iv)
        shareIv = findViewById(R.id.desc_share_iv)
        picIv = findViewById(R.id.desc_iv_pic)
        titleTv = findViewById(R.id.desc_tv_title)
        contentTv = findViewById(R.id.desc_tv_content)

        backIv.setOnClickListener(this)
        shareIv.setOnClickListener(this)

        val id = intent.getStringExtra("hisId")
        presenter = HistoryDescPresenter(this)
        id?.let { presenter?.loadGet(it) }
    }

    override fun onDestroy() {
        super.onDestroy()
        presenter?.onDestroy()
    }

    override fun onClick(v: View?) {
        when (v?.id) {
            R.id.desc_back_iv -> finish()
            R.id.desc_share_iv -> shareContent()
        }
    }

    private fun shareContent() {
        val text = if (resultBean != null) {
            "想要了解${resultBean!!.title}详情么？快来下载历史上的今天App吧！"
        } else {
            "我发现一款好用的软件-历史上的今天，快来一起探索这个APP吧！"
        }
        val intent = Intent(Intent.ACTION_SEND).apply {
            type = "text/plain"
            putExtra(Intent.EXTRA_TEXT, text)
        }
        startActivity(Intent.createChooser(intent, "历史上的今天"))
    }

    override fun showResult(bean: HistoryDescBean) {
        if (!bean.result.isNullOrEmpty()) {
            resultBean = bean.result[0]

            Log.d(TAG, resultBean.toString())

            titleTv.text = resultBean!!.title
            contentTv.text = resultBean!!.content

            val picUrl = resultBean!!.picUrl
            if (!picUrl.isNullOrEmpty()) {
                val firstUrl = picUrl[0].url
                if (!TextUtils.isEmpty(firstUrl)) {
                    picIv.visibility = View.VISIBLE
                    Picasso.get().load(firstUrl).into(picIv)
                } else {
                    picIv.visibility = View.GONE
                }
            } else {
                picIv.visibility = View.GONE
            }
        } else {
            titleTv.text = "暂无数据"
            contentTv.text = "未找到相关历史事件详情"
            picIv.visibility = View.GONE
        }
    }
}
